framework/src/main/java/org/checkerframework/framework/util/SimpleQualifierHierarchy.java

```diff
@@ -0,0 +1,239 @@
+    }
+
+    /**
+     * Creates and returns a mapping from qualifier kind to an annotation mirror created from the
+     * qualifier kind's annotation class.
+     *
+     * @param elements element utils
+     * @return a mapping from qualifier kind to its annotation mirror
+     */
+    @RequiresNonNull("this.qualifierKindHierarchy")
+    protected Map<QualifierKind, AnnotationMirror> createAnnotationMirrors(
+            @UnderInitialization SimpleQualifierHierarchy this, Elements elements) {
+        Map<QualifierKind, AnnotationMirror> quals = new TreeMap<>();
+        for (QualifierKind kind : qualifierKindHierarchy.allQualifierKinds()) {
+            if (kind.hasElements()) {
+                throw new TypeSystemError(
+                        "SimpleQualifierHierarchy cannot be used with annotations that have elements. Found %s: ",
+                        kind);
+            }
+            quals.put(kind, AnnotationBuilder.fromClass(elements, kind.getAnnotationClass()));
+        }
+        return Collections.unmodifiableMap(quals);
+    }
+
+    /**
+     * Creates and returns the unmodifiable set of top {@link AnnotationMirror}s.
+     *
+     * @return the unmodifiable set of top {@link AnnotationMirror}s
+     */
+    @RequiresNonNull("this.kindToAnnotationMirror")
+    protected Set<AnnotationMirror> createTops(@UnderInitialization SimpleQualifierHierarchy this) {
+        Set<AnnotationMirror> tops = AnnotationUtils.createAnnotationSet();
+        for (Map.Entry<QualifierKind, AnnotationMirror> entry : kindToAnnotationMirror.entrySet()) {
+            if (entry.getKey().isTop()) {
+                tops.add(entry.getValue());
+            }
+        }
+        return Collections.unmodifiableSet(tops);
+    }
+
+    /**
+     * Creates and returns the unmodifiable set of bottom {@link AnnotationMirror}s.
+     *
+     * @return the unmodifiable set of bottom {@link AnnotationMirror}s
+     */
+    @RequiresNonNull("this.kindToAnnotationMirror")
+    protected Set<AnnotationMirror> createBottoms(
+            @UnderInitialization SimpleQualifierHierarchy this) {
+        Set<AnnotationMirror> bottoms = AnnotationUtils.createAnnotationSet();
+        for (Map.Entry<QualifierKind, AnnotationMirror> entry : kindToAnnotationMirror.entrySet()) {
+            if (entry.getKey().isBottom()) {
+                bottoms.add(entry.getValue());
+            }
+        }
+        return Collections.unmodifiableSet(bottoms);
+    }
+
+    /**
+     * Returns the {@link QualifierKind} for the given annotation.
+     *
+     * @param anno an annotation that is a qualifier in this
+     * @return the {@code QualifierKind}
+     */
+    protected QualifierKind getQualifierKind(AnnotationMirror anno) {
+        String name = AnnotationUtils.annotationName(anno);
+        QualifierKind kind = qualifierKindHierarchy.getQualifierKind(name);
+        if (kind == null) {
+            throw new BugInCF("Annotation not in hierarchy: %s", anno);
```

Comment:
Is this a `TypeSystemError`?

Reply:
The framework itself could call this method with an incorrect annotation, so it could be a bug in the framework.